recommence/_utils/pickle.py

```diff
@@ -0,0 +1,6 @@
+import pickle
+
+
+def read_pickle(path: str):
```

Comment:
This is just a bit silly --- smaller functionality than I would typically break into its own function. The motivation here is that I really wanted to avoid deeply nested code blocks here: https://github.com/Amii-Open-Source/recommence/pull/10/files#diff-5578bfb8a3cbdf5faf84cc81663b09c01a91f43661fc2f7c6b6099ba2fa3a511R64

I want the "business" logic to read very linearly:
1. Check if an uncompressed checkpoint exists. If so, load it
2. Check if a compressed checkpoint exists. If so, load it
3. Give up looking for checkpoints